tests/remotes.py

```diff
@@ -141,10 +142,13 @@ class GDrive:
     def should_test():
         return os.getenv(GDriveRemote.GDRIVE_CREDENTIALS_DATA) is not None
 
-    def get_url(self):
-        if not getattr(self, "_remote_url", None):
```

Comment:
removing this will break `TestGDriveRemoteCLI`  most likely

Reply:
in general I believe we should probably introduce caching of get_url in for other remotes as well - it is strange to return different values within one test

Reply:
@shcheklein It actually shouldn't, because other remotes do that too and just reuse the results. Need to check maybe there is something wrong or special about gdrive...

Reply:
@efiop I think they just call it once, in case of GDrive it just happened that we need it in multiple places. And in general, it's def not a good and safe interface - this value should be cached one way or another and stay the same during the test.